src/server/scripts/Outland/BlackTemple/instance_black_temple.cpp

```diff
@@ -458,6 +458,27 @@ class spell_black_temple_dementia_aura : public AuraScript
     }
 };
 
+// 39649 - Summon Shadowfiends
+class spell_black_temple_summon_shadowfiends : public SpellScript
+{
+    PrepareSpellScript(spell_black_temple_summon_shadowfiends);
+
+    void HandleScriptEffect(SpellEffIndex /*effIndex*/)
+    {
+        Unit* caster = GetCaster();
+        if (!caster)
+            return;
+
+        for (uint8 i = 0; i < 11; i++)
+            caster->CastSpell(caster, SPELL_SUMMON_SHADOWFIENDS, true);
```

Comment:
Is a spell validation required for this?

Reply:
Habit, otherwise it would be if (Unit* caster= GetCaster()), same thing, at least for me.

Reply:
I know what validate is, just most of the SPELL_EFFECT_SCRIPT_EFFECT don't use it, even if they do cast a spell.